app/controllers/report_controller.rb

```diff
@@ -58,7 +58,36 @@ def add_record
   end
 
   def raw_data
-    render json: Report.all
+    reports = Report.order(:max)
+    response = {}.tap do |res|
+      res[:header] = {}.tap do |h|
+        h[:user] = 'User'
+        h[:platform] = 'Platform'
+        h[:location] = 'Location'
+        h[:core_number] = 'NO. cores'
+        h[:ram] = 'RAM(GB)'
+        h[:main] = 'Equivalent CO2 emissions per core at full load (kgCO2eq/h)'
+      end
+      rank = 1
+      res[:reports] = reports
+      .group_by(&:max)
+      .values.flat_map do |rep_group|
+        current_rank = rank
+        rank += rep_group.size
+        rep_group.map do |rep|
+          {}.tap do |new_rep|
+            new_rep[:rank] = current_rank
+            new_rep[:user] = User.where('id = ?', rep[:user_id]).pluck(:username).first
```

Comment:
Can use `find` rather than `where` and `first`

Reply:
Good knowledge. I feel this should be greatly changed/expanded by Matt in the future so it might not be quite necessary to change atm.